### PR TITLE
This branch is to connect Bookstore API to Bookstore project

### PR DESCRIPTION
**Pull request by steveWDamesJr**
## Connect Bookstore API

-Add Redux-Thunk
-Add Redux-Devtools-extension
-Use axios for making API calls
-Add Action Creators
-Fetch books
-Add action type to dispatch to pass API response to reducer and update state
-Refactor addBook and removeBook to persist changes in server

-Debug linters

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,40 +1,34 @@
...
-}
+export const addBook = (book) => async (dispatch) => {
+  await BookStoreUtility.apiToAddBook(book);
+  dispatch({ type: ADD_BOOK, book });
+};
+
+export const removeBook = (id) => async (dispatch) => {
+  await BookStoreUtility.apiToRemoveBook();
+  dispatch({ type: REMOVE_BOOK, id });
+};
```
Comment (brytebee) [inline src/redux/books/books.js]: - [ ] **Required:** You have done great work on this milestone. Well done 👏👏

However, I have observed that your remove doesn't remove books from the API. Kindly fix this behaviour.

Remove_from_api

File: src/index.js [hunk 2]
```diff
@@ -1,13 +1,19 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 import { Provider } from 'react-redux';
+import { BrowserRouter as Router } from 'react-router-dom';
 import store from './redux/configureStore';
+// import { getBooks } from './redux/books/books';
 import './index.css';
 import App from './App';
 
+// store.dispatch(getBooks);
```
Comment (brytebee) [inline src/index.js]: - [x] **Required:** You have left some commented code in your codebase. It is unprofessional to commit commented code. Kindly remove them throughout your codebase.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @[steveWDamesJr](https://github.com/steveWDamesJr)👋,

Good job so far coming so far with the progress you've made! 💪💪

## Highlights
You have done amazing in the following:
- Professional looking documentation.✔️
- An action Creator for fetching books is created.✔️
- An action Creator for adding/posting books is created.✔️
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @steveWDamesJr 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional suggestions
- Please fix your linters

Cheers and Happy coding!👏👏👏


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**